### PR TITLE
Disp 139/input

### PR DESCRIPTION
**Pull request by alatielle**
## What

implemented TextField with a possibility to render Inputs and Textareas as the field

### Media

Screenshot 2024-09-18 at 13 07 21
Screenshot 2024-09-18 at 13 07 32
Screenshot 2024-09-18 at 13 07 46
Screenshot 2024-09-18 at 13 07 57

Screenshot 2024-09-18 at 13 10 27

## Why

Closes [DISP-139](https://jira.intapps.it/browse/DISP-139)

### CODE REVIEW

File: src/components/experimental/TextField/ClearButton.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { XCrossCircleIcon } from '../../../icons';
+
+const StyledButton = styled(BaseButton)`
+    appearance: none;
+    background: none;
+    border: none;
+    display: flex;
+    margin: 0;
+    padding: 0;
+    color: ${getSemanticValue('on-surface-variant')};
```
Comment (arturmiglio) [inline src/components/experimental/TextField/ClearButton.tsx:14]: What does the `-variant` suffix mean?

File: src/components/experimental/TextField/Field.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    resize: none;
+    min-height: ${textStyles.variants.body1.lineHeight};
+`;
+
+export const Input = styled(BaseInput)`
+    ${fieldStyles}
+
+    &[type='search'] {
+        &::-webkit-search-cancel-button {
```
[the remark this answers is not in the document]
Comment (alatielle) [reply src/components/experimental/TextField/Field.ts:36]: Thought about IE, but... is it alive? 💀

File: src/components/experimental/TextField/TextField.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+`;
+
+const Counter = styled.span`
+    grid-area: counter;
+`;
+
+const Wrapper = styled(BaseTextField)`
+    padding: ${get('space.2')} ${get('space.0')};
+
+    &[data-disabled] {
```
[the remark this answers is not in the document]
Comment (alatielle) [reply src/components/experimental/TextField/TextField.tsx:114]: These are recommended by the react-aria library for styling

File: src/components/experimental/TextField/Label.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    position: absolute;
+    top: 50%;
+    color: currentColor;
+
+    ${textStyles.variants.body1}
+
+    transform: translate3d(0, calc(-${textStyles.variants.body1.lineHeight} / 2), 0);
+    transform-origin: 0;
+
+    transition: top 200ms ease, font-size 200ms ease, transform 200ms ease;
```
[the remark this answers is not in the document]
Comment (alatielle) [reply src/components/experimental/TextField/Label.ts:22]: CSS variables or move it to the theme, as we do for border radii and stuff?

File: src/components/experimental/TextField/Field.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (arturmiglio) [reply src/components/experimental/TextField/Field.ts:36]: According to the docs we give full support for the last 2 versions of: Chrome, **Firefox**, **Safari**, and Edge.

So I guess we would need special styling (or least check support) for Firefox and Safari.

Note: maybe we are using some kind of CSS post-processor that can be leveraged here?

https://wave.free-now.com/?path=/docs/browser-support--docs

File: src/components/experimental/TextField/Label.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (arturmiglio) [reply src/components/experimental/TextField/Label.ts:22]: Both should work...

CSS Variable would be more "easy to access" (no get function) but less type safe (stylelint should help, tho).

But theme prop would be simpler to maintain(?) + type safe.

File: src/components/experimental/TextField/ClearButton.tsx (hunk 1 above)
Comment (arturmiglio) [reply src/components/experimental/TextField/ClearButton.tsx:14]: Hmm, like an alternate version.... 

I'd talk about this with Marian... it seems that meaning is a bit subjective. Probably there's an explanation... but seems that the explanation would be needed, if someone would autonomously combine this colours (without a design spec sheet).

**Review by arturmiglio — APPROVED** [review]

File: src/components/experimental/TextField/Label.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (alatielle) [reply src/components/experimental/TextField/Label.ts:22]: I'll create a separate task for this: the designs use the CSS variables for shapes and sizes

File: src/components/experimental/TextField/Field.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alatielle) [reply src/components/experimental/TextField/Field.ts:36]: Safari is also webkit and firefox does not add a cross icon to input type search

**Pull request merged**